### ISSUE TITLE
gun unable to handle continuous writes 

### ISSUE THREAD

**Issue by zer09** [amark/gun#592]
Hello just doing gun test again.

Just tried to do continuous writes on gun on a loop, with just this.
```
  let rands = global.store.get('randomstrings');

  for (let i = 0; i < 100000; i++) {
    let rnd = randomstring.generate({
      length: 7
    });

    rands.get(rnd).path('exists').once(v => {
      if (!v) {
        let rndStore = global.store.get(rnd);
        rndStore.put({
          exists: true
        });

        rands.set(rndStore)
        console.log({
          rnd,
          i
        })
      }
    });
  }

```

here is the sample repo. https://github.com/zer09/gunwritetest

but executing that, node will crash with this log.
```
<--- Last few GCs --->

[7118:0x55bd130f1840]   238610 ms: Mark-sweep 1158.0 (1535.7) -> 1157.9 (1538.2) MB, 3332.7 / 0.2 ms  allocation failure GC in old space requested
[7118:0x55bd130f1840]   242101 ms: Mark-sweep 1157.9 (1538.2) -> 1157.8 (1491.2) MB, 3487.0 / 0.1 ms  last resort GC in old space requested
[7118:0x55bd130f1840]   245559 ms: Mark-sweep 1157.8 (1491.2) -> 1157.8 (1483.2) MB, 3458.7 / 0.1 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x40f60425879 <JSObject>
    1: track [/home/gconsellado/node/guntest/node_modules/gun/gun.js:~630] [pc=0x3b8135ac10e0](this=0x1d5f855a5e79 <Object map = 0x308bee15029>,id=0x3f0e551cb01 <String[11]: c6wlo71f63m>,pass=0x357bbea822d1 <undefined>)
    2: arguments adaptor frame: 1->2
    3: next(aka root) [/home/gconsellado/node/guntest/node_modules/gun/gun.js:706] [bytecode=0x129fb14cabb1 offset=121](this=0x1d5f855a5f11 <O...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x55bd101bc312 [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewFixedArray(int, v8::internal::PretenureFlag) [node]
 6: v8::internal::HashTable<v8::internal::NameDictionary, v8::internal::NameDictionaryShape>::New(v8::internal::Isolate*, int, v8::internal::PretenureFlag, v8::internal::MinimumCapacity) [node]
 7: v8::internal::HashTable<v8::internal::NameDictionary, v8::internal::NameDictionaryShape>::EnsureCapacity(v8::internal::Handle<v8::internal::NameDictionary>, int, v8::internal::PretenureFlag) [node]
 8: v8::internal::Dictionary<v8::internal::NameDictionary, v8::internal::NameDictionaryShape>::Add(v8::internal::Handle<v8::internal::NameDictionary>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyDetails, int*) [node]
 9: v8::internal::LookupIterator::ApplyTransitionToDataProperty(v8::internal::Handle<v8::internal::JSObject>) [node]
10: v8::internal::Object::AddDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow, v8::internal::Object::StoreFromKeyed) [node]
11: v8::internal::Object::SetProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::LanguageMode, v8::internal::Object::StoreFromKeyed) [node]
12: v8::internal::Runtime::SetObjectProperty(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::LanguageMode) [node]
13: v8::internal::Runtime_SetProperty(int, v8::internal::Object**, v8::internal::Isolate*) [node]
14: 0x3b81357842fd
[1]    7108 abort (core dumped)  npm start

```

**Comment by fuchidahiro** [comment]
@zer09 This is not a gun issue per se, but rather node.js heap memory being overwhelmed by the huge amount of async calls you create when running a loop of writes. It's very easy to overload a node server that way and the recommended write pattern (in my experience) is either batching writes with breaks in between for nodes GC to clean up and open up memory or to have each loop execution wait a bit.

**Comment by amark** [comment]
@zer09 @Dletta I'd still like to make sure (if possible) we address this. There should be someway, especially since GUN does a lot of deferred processing anyways.

**Comment by bugs181** [comment]
Does `process.nextTick()` solve this issue? Ref: https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/

**Comment by Dletta** [comment]
It could, but what is the browser equivalent of it? @bugs181

**Comment by bugs181** [comment]
> It could, but what is the browser equivalent of it? @bugs181

As discussed in Gitter (moving here for posterity), `nextTick` can be polyfilled. I believe this is happening because the callback queue is getting filled before we can flush to I/O. The solution is to build an iterator that can pause the Event queue for I/O, before moving back to user-land.
